guide/message-components/buttons.md

```diff
@@ -0,0 +1,172 @@
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="user" :command="true">buttons</DiscordInteraction>
+		</template>
+		<template #actions>
+			<DiscordButtons>
+				<DiscordButton type="primary">Primary</DiscordButton>
+				<DiscordButton type="secondary">Secondary</DiscordButton>
+				<DiscordButton type="success">Success</DiscordButton>
+				<DiscordButton type="danger">Danger</DiscordButton>
+				<DiscordButton type="link" url="https://discord.js.org">Link</DiscordButton>
+			</DiscordButtons>
+		</template>
+	</DiscordMessage>
+</DiscordMessages>
+
+- `Primary` style buttons are blue. These are suitable for most general purpose actions, where it's the primary or most significant action expected.
+- `Secondary` style buttons are grey. Use these for less important actions like the "Cancel" button in the example above.
+- `Success` style buttons are green. Similar to the Primary button, these are a good choice for "positive" confirmation actions.
+- `Danger` style buttons are red. Where the action being confirmed is "destructive", such a ban or delete, using a red button helps alert the user to the risk of the action.
+- `Link` style buttons are also grey, but are tagged with the "external link" symbol. These buttons will open the provided link in the browser without sending an interaction to the bot.
+
+## Link buttons
+
+Link buttons are a little different to the other styles. `Link` buttons _must_ have a `url`, _cannot_ have a `customId` and _do not_ send an interaction event when clicked.
+
+```js{3}
+const button = new ButtonBuilder()
+	.setLabel('discord.js docs')
+	.setURL('https://discord.js.org')
+	.setStyle(ButtonStyle.Link);
+```
+
+## Disabled buttons
+
+If you want to prevent a button from being used, but not remove it from the message, you can disable it with the <DocsLink section="builders" path="ButtonBuilder:Class#setDisabled" type="method"/> method:
+
+```js {5}
+const button = new ButtonBuilder()
+	.setCustomId('disabled')
+	.setLabel('Click me?')
+	.setStyle(ButtonStyle.Primary)
+	.setDisabled(true);
+```
+
+<DiscordMessages>
+	<DiscordMessage profile="bot">
+		<template #interactions>
+			<DiscordInteraction profile="user" :command="true">button</DiscordInteraction>
+		</template>
+		Are you even able to 
+		<template #actions>
+			<DiscordButtons>
+				<DiscordButton :disabled="true">Click me?</DiscordButton>
+			</DiscordButtons>
+		</template>
+	</DiscordMessage>
+</DiscordMessages>
+
+## Emoji buttons
+
+If you want to use a guild emoji within a <DocsLink path="class/ButtonBuilder"/>, you can use the <DocsLink path="ButtonBuilder:Class#setEmoji" type="method"/> method:
```

Comment:
This one should reference our overridden method, as we're making use of providing a snowflake.

```suggestion
If you want to use a guild emoji within a <DocsLink path="class/ButtonBuilder"/>, you can use the <DocsLink path="class/ButtonBuilder?scrollTo=setEmoji" type="method"/> method:
```